client/my-sites/themes/controller/index.node.js

```diff
@@ -42,27 +40,14 @@ export function fetchThemeDetailsData( context, next ) {
 	}
 
 	const themeSlug = context.params.slug;
-	const theme = themeDetailsCache.get( themeSlug );
+	const theme = getThemeDetails( context.store.getState(), themeSlug );
 
-	if ( theme ) {
+	if ( Object.keys( theme ).length ) {
```

Comment:
Should we give the theme a time-to-live, or maybe do something with `date_updated`? As it stands this will rely on a Calypso re-deploy to flush the cache.